### PR TITLE
Sticky top banner race conditions

### PR DESCRIPTION
**Pull request by rich-nguyen**
There are some errors being thrown in Sentry which look like sticky top banner is operating on a dom tree without a `top-above-nav` slot. This is likely to be a race condition, where the commercial app is in the process of closing (ie removing) server-rendered slots. So we need to wait for commercial dom writes to finalise the slot structure on the page before running any sticky top banner code. I've moved the whole module into the commercial app.

Also, the logic to decide if we run with a sticky top banner, a top banner, or no top banner and header, was all a bit crazy. I've moved it to the server and tried to break it down, but it's still fundamentally complicated.

### CODE REVIEW

File: static/src/javascripts/bootstraps/enhanced/common.js [hunk 1]
```diff
@@ -133,27 +131,6 @@ define([
...
-                if (!(config.switches.disableStickyAdBannerOnMobile && detect.getBreakpoint() === 'mobile')
-                    && !config.page.shouldHideAdverts
-                    && config.page.section !== 'childrens-books-site'
-                    && !config.tests.abNewHeaderVariant
-                    && (config.page.hasSuperStickyBanner
-                        || config.page.contentType !== 'Interactive'
-                        && config.page.contentType !== 'Crossword'
-                        && config.page.contentType !== 'Hosted'
-                        && !config.page.isImmersive
-                        && !config.page.isUsMinute
```
Comment (rich-nguyen) [inline static/src/javascripts/bootstraps/enhanced/common.js]: i don't think this key ever exists, @gtrufitt, its called `isMinuteArticle` client-side?

File: static/src/javascripts/projects/common/modules/commercial/sticky-top-banner.js [hunk 2]
```diff
@@ -211,7 +211,9 @@ define([
     };
 
     var initialise = function () {
-        if (detect.isBreakpoint({ min: 'desktop' })) {
+        // Although we check as much config as possible to decide whether to run sticky-top-banner,
+        // it is still entirely possible for the ad slot to be closed.
+        if (detect.isBreakpoint({ min: 'desktop' }) && $adBannerInner[0]) {
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/commercial/sticky-top-banner.js]: Hopefully this is the catch-all line which stops all the errors flooding sentry. But the rest of the PR is about making sure the ad slot really is there.

File: static/src/javascripts/bootstraps/commercial.js [hunk 3]
```diff
@@ -46,12 +50,36 @@ define([
         ['cm-sliceAdverts', sliceAdverts.init],
         ['cm-frontCommercialComponents', frontCommercialComponents.init],
         ['cm-closeDisabledSlots', closeDisabledSlots.init]
+    ],
+    secondaryModules = [
```
Comment (JonNorman) [inline static/src/javascripts/bootstraps/commercial.js]: I've been told we are moving away from single-var declaration, but I don't feel strongly either way :)

**Comment by JonNorman** [comment]
Looks good to me, pending whether `isMinuteArticle` is still a thing :+1:

File: static/src/javascripts/bootstraps/enhanced/common.js (hunk 1 above)
Comment (gtrufitt) [inline static/src/javascripts/bootstraps/enhanced/common.js]: Yeah, `isMinuteArticle`, probably a missed refactor somewhere along the line

**Comment by rich-nguyen** [comment]
it's a thing, i used the server-side equivalent.

**Pull request merged**